code/datums/components/mechComp_signals.dm

```diff
@@ -297,7 +297,8 @@
 
 //If it's a multi-tool, let the user configure the device.
 /datum/component/mechanics_holder/proc/attackby(var/comsig_target, obj/item/W as obj, mob/user)
-	if(!ispulsingtool(W) || !isliving(user) || user.stat)
+	var/obj/machinery/door/hacked_door = comsig_target
```

Comment:
It's always a door?

Reply:
It always exists and is passed in?

Reply:
comsig_target is the mechcomp component, could be any component
it's a workaround for the case when the component is a door with an open panel
can't really make it work via istype due to how the whole proc is coded so I did it that way :iiam:

Reply:
yea, I realized what I did now, oops
now the windows works for things other than doors aswell